### ISSUE TITLE
Internal Link List causes problems when injecting pages

### ISSUE THREAD

**Issue by KiraResari** [michitux/dokuwiki-plugin-move#151]
I used to have the following page hierarchy:

internal:start > internal:customers:start

However, after working on it for a bit, I realized that what I needed was an extra page between internal:start and internal:customers:start, so here is what I did:

1. I renamed internal:customers:start to internal:customers:overview
2. I created a new internal:customers:start page

This is when the following project emerged:

* On internal:start, I now have a link to internal:customers:overview
* When I try to rename it to internal:customers:start, nothing happens, and it is still linking to internal:customers:overview
* Looking at the source files on the server, I see that the internal/start.txt file says "internal:customers:start" for the link
* Also, when I disable the move-plugin, the link is correctly displayed and leads to internal:customers:start (but other links break at the same time)

I assume this happens because when renaming a page, the move plugin does not actually adjust the links in the files, but rather creates a link-map somewhere. I am sure that was done for a reason, but it leads to easily reproducible problems like this one.

Is there any way around this?

To sum it up, here steps by which this can be reproduced:

1. Create a link on Page A to Page B
2. Rename Page B to Page C
3. Create a new Page B
4. Try to change the link on Page A from "Page C" to "Page B"

**Comment by phy25** [comment]
I cannot reproduce this on move 2018-04-30 and DokuWiki 2018-04-22a "Greebo". Maybe only sometimes it doesn't work (it might be that the move lock is not released?).

Thanks to your issue, I just realize that `move != rename`. It's hard to do automatic link updating easily, so what this plugin do is to apply changes later when rendering (viewing by webpage). If you are curious why it's hard, see the older, retired plugin: http://www.dokuwiki.org/plugin:pagemove

**Comment by bbits7** [comment]
We are having the same issue... there must be someway to 'stop' renaming...

**Comment by phy25** [comment]
@bbits7 As I mentioned, "what this plugin do is to apply changes later when rendering (viewing by webpage)". So you may want to look at the page's metadata to remove the redirect. I am not sure if there's any existing tool to edit the metadata on the webpage, but the format of metadata file is PHP's serialized object. Take your own risk when modifying the file.

In the meantime, if you can provide a reproducible testcase, developers will be able to test it and fix the hidden bug.
